route/route.go

```diff
@@ -626,3 +646,119 @@ func getFirstValueFromMetadata(key string, md metadata.MD) string {
 	}
 	return ""
 }
+
+type environmentCache struct {
```

Comment:
It probably doesn't matter because this is only called in one place, but if this were to be used in multiple places, I'd rather see the getFn passed in to the environmentCache on construction, and then called with the key when the value is not found or the ttl expires. Then the cache basically looks like a key/value map. 

However, the way you've done it here is not wrong and I'm not requesting you to change it.

Reply:
Thanks for the feedback - I like your suggestion as it simplifies the usage fo the cache so dod the update.